### DISCUSSION TITLE
Sending one component data to anaother

### DISCUSSION THREAD

**Discussion by adinath1233** [deepdoctection/deepdoctection#356] [Q&A] [answered]
Hello I need to ask how I can send the data or output from one component to another like I have a pipeline 
e.g.: 
pipe = dd.DoctectionPipe([image_layout, text_extract])

which has an image layout service and text extraction service

is data or output of the image layout service sent to the text extraction service?
is is sent by Doctection pipe or need to configure it?

**Comment by JaMe76** [comment]
Each pipeline component processes its data using data flows (e.g. generators). Once the pipeline component is instantiated 
it can take data from previous dataflow, invokes its predictors/function and pass the outcome to the next generator. 

The pipeline is responsible for processing data through every component. What the pipeline is doing under the hood is concatenating all components to a dataflow:

```python
df = # some incoming dataflow
df = image_layout.predict_dataflow(df)
df = text_extract.predict_dataflow(df)
...
```

The reference code is here:

https://github.com/deepdoctection/deepdoctection/blob/661185af495063909d086bd98a3df566c051911f/deepdoctection/pipe/base.py#L227

Does this answer your question?

**Reply by adinath1233** [reply]
Thank you @JaMe76 so I do not need to do anything it will be passed by dataflow am I right? So If I layout data is predicted using predict and I want to send that data ahead to text extracted it will send that ahead automatically ?

**Reply by JaMe76** [reply, marked as answer]
Once you use a pipeline, e.g. DoctectionPipe it will take care of passing the data forward to the next component, yes. The order for running pipeline components is determined by their position in the pipeline_component_list.

**Reply by adinath1233** [reply]
Thank You @JaMe76 for responding apreicated.
